dowhy/gcm/model_evaluation.py

```diff
@@ -0,0 +1,794 @@
+                )
+
+        if self.overall_kl_divergence is not None:
+            summary_strings.append("\n==== Evaluation of Generated Distribution ====")
+            summary_strings.append(
+                "The overall average KL divergence between the generated and observed distribution is %s"
+                % self.overall_kl_divergence
+            )
+            summary_strings.append(_get_kl_divergence_interpretation_string(self.overall_kl_divergence))
+
+        if self.graph_falsification is not None:
+            summary_strings.append("\n==== Evaluation of the Causal Graph Structure ====")
+            summary_strings.append(str(self.graph_falsification))
+            if self.plot_falsification_histogram:
+                plot_evaluation_results(self.graph_falsification)
+
+        summary_strings.append("\n==== NOTE ====")
+        summary_strings.append(
+            "Always double check the made model assumptions with respect to the graph structure and "
+            "choice of causal mechanisms."
+        )
+        summary_strings.append(
+            "All these evaluations give some insight into the goodness of the causal model, but should not be "
+            "overinterpreted, since some causal relationships can be intrinsically hard to model. Furthermore, many "
+            "algorithms are fairly robust against misspecifications or poor performances of causal mechanisms."
+        )
+
+        return "\n".join(summary_strings)
+
+
+def evaluate_causal_model(
+    causal_model: ProbabilisticCausalModel,
+    data: pd.DataFrame,
+    max_num_samples: int = -1,
+    evaluate_causal_mechanisms: bool = True,
+    compare_mechanism_baselines: bool = False,
+    evaluate_invertibility_assumptions: bool = True,
+    evaluate_overall_kl_divergence: bool = True,
+    evaluate_causal_structure: bool = True,
+    config: Optional[EvaluateCausalModelConfig] = None,
+) -> CausalModelEvaluationResult:
+    """Evaluates the given causal model by running different evaluations.
+
+    --- Evaluation of Causal Mechanisms ---
+    The quality of the causal mechanisms is assessed using k-fold cross validation. This means that the models are trained
+    from scratch multiple times, which might take a significant amount of time for larger models. Within each fold, the models
+    are assessed using the continuous ranked probability score (CRPS). Optionally, the mechanisms are compared with
+    baseline models to see if they are performing significantly better or equally good.
+
+    --- Evaluation of Invertible Functional Causal Model Assumption ---
+    Invertible causal mechanisms rely on the assumption that the inputs are independent of the reconstructed noise.
+    This is, assuming there are no hidden confounders, the noise should be independent of parents of a node. This
+    can be evaluated by testing statistical independence between the reconstructed noise and the used input samples.
```

Comment:
Have you checked that this also looks pretty when rendered in the documentation?